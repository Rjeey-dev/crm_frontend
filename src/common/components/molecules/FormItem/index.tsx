import React from 'react';

import Field from "common/components/atoms/Field/index";

interface IProps {
    meta: any,
    input: any,
}

const FormItem = ({ meta, input, ...props }: IProps) => {
    const fieldProps = {...props, initial: meta.initial, ...input, invalid: meta.touched && !!meta.error, error: meta.error,};

    if (fieldProps.type === 'text' || fieldProps.type === 'textarea') {
        return <Field {...fieldProps} >{fieldProps.children}</Field>
    }

    return <Field {...fieldProps} >{fieldProps.children}</Field>
};

export default FormItem;
