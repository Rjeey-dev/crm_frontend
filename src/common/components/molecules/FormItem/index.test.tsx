import {shallow} from "enzyme";
import React from 'react';

import FormItem from './index';

it("Renders <FormItem/> with correct properties with wrapper", () => {
    const output = shallow(
        <FormItem meta={{touched: true, error: 'error'}} input={{}}/>
    ).html();

    expect(output).toMatchSnapshot();
});