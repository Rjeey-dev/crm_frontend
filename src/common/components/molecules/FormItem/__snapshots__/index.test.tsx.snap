// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders <FormItem/> with correct properties with wrapper 1`] = `"<div class=\\"form-group\\"><div class=\\"input-group\\"><input aria-describedby=\\"undefinedError\\" error=\\"error\\" class=\\"form-control form-control-danger\\"/><div class=\\"input-group-append\\"><span class=\\"input-group-text input-group-text-danger\\"><i class=\\"mdi mdi-check-circle with-error\\"></i></span></div></div><label id=\\"undefinedError\\" class=\\"error mt-2 text-danger\\">Missing translationId:  for language: \${ languageCode }</label></div>"`;
